src/server/game/Movement/MotionMaster.cpp

```diff
@@ -322,7 +323,7 @@ void MotionMaster::Add(MovementGenerator* movement, MovementSlot slot/* = MOTION
         return;
     }
 
-    if (HasFlag(MOTIONMASTER_FLAG_UPDATE))
+    if (HasFlag(MOTIONMASTER_FLAG_UPDATE) || (!_owner->IsInWorld() && !_defaultInitialized))
```

Comment:
Creatures in formations call MoveIdle() instead of Initialize(), causing _defaultInitialized to never be set to true.